### PR TITLE
fix weird lockfile irregularities

### PR DESCRIPTION
**Pull request by brad-decker**
## Explanation
While working on #20249 I hit some weird irregularities with our lockfile that included github resolutions that didn't make sense. I deleted those entries from the lockfile and regenerated with yarn. This will help my yarn branch from constantly getting locked with yarn.lock merge conflicts, which for some reason always prefer the github URLs from the original instead of the new correct ones <_<

### CODE REVIEW

**Comment by brad-decker** [comment]
@metamaskbot update-policies

**Comment by brad-decker** [comment]
@Gudahtt looks like the policy update failure is not specific to yarn v4 but rather something I changed in the lockfile.

File: yarn.lock [hunk 1]
```diff
@@ -14744,11 +14746,11 @@ __metadata:
   linkType: hard
 
 "dot-prop@npm:^4.1.0":
-  version: 5.2.0
-  resolution: "dot-prop@npm:5.2.0"
+  version: 4.2.1
```
[the remark this answers is not in the document]
Comment (brad-decker) [reply yarn.lock:14749]: The override in that pr was for this: https://github.com/advisories/GHSA-ff7x-qrg7-qggm

we currently resolve to 4.2.1 which is the fixed version of 4. also the resolution in that PR was for a dependency from ethers which is no longer the reason we have dot-prop

File: yarn.lock [hunk 2]
```diff
@@ -16165,11 +16167,11 @@ __metadata:
 
 "ethereumjs-abi@npm:0.6.8, ethereumjs-abi@npm:^0.6.4, ethereumjs-abi@npm:^0.6.8":
   version: 0.6.8
-  resolution: "ethereumjs-abi@https://github.com/ethereumjs/ethereumjs-abi.git#commit=ee3994657fa7a427238e6ba92a84d0b529bbcde0"
+  resolution: "ethereumjs-abi@npm:0.6.8"
```
[the remark this answers is not in the document]
Comment (brad-decker) [reply yarn.lock:16170]: was it a manual override to the lockfile or something that was generated? All I see in that diff from a package perspective is metamask controllers being updated.

**Comment by Gudahtt** [comment]
Likely the policy update OOM error isn't related to Node.js v18, because it seems to work fine on other PRs that are using Node.js v18 (e.g. [here](https://github.com/MetaMask/metamask-extension/actions/runs/5719958173/job/15499050442))

File: yarn.lock (hunk 2 above)
[the remark this answers is not in the document]
Comment (Gudahtt) [reply yarn.lock:16170]: Looks like this was brought in from an ancient version of `eth-sig-util`:

```


eth-sig-util@^1.4.2:
  version "1.4.2"
  resolved "https://registry.yarnpkg.com/eth-sig-util/-/eth-sig-util-1.4.2.tgz#8d958202c7edbaae839707fba6f09ff327606210"
  integrity sha1-jZWCAsftuq6Dlwf7pvCf8ydgYhA=
  dependencies:
    ethereumjs-abi "git+https://github.com/ethereumjs/ethereumjs-abi.git"
    ethereumjs-util "^5.1.1"
```

It looks like the git reference [wasn't used for any documented reason](https://github.com/MetaMask/eth-sig-util/pull/5) (maybe it wasn't yet published at the time? Probably just a mistake though).

**Comment by Gudahtt** [comment]
@metamaskbot update-policies

**Comment by Gudahtt** [comment]
I have updated the policies locally (using Linux, to ensure they match the CI policies), and started another policy update just to see if it still fails. Likely it will.

**Comment by brad-decker** [comment]
Applied learnings from #20337 to this PR and hoping for similar results. I have not yet copied @Gudahtt 's overrides for UInt8Array so I may have to do that in addition but want to check this first.

**Comment by brad-decker** [comment]
@metamaskbot update-policies

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Review by darkwing — APPROVED** [review]

**Pull request merged**